java/src/main/java/lesson03/solution/Hello.java

```diff
@@ -28,22 +28,28 @@ private Hello(Tracer tracer) {
 
     private String getHttp(int port, String path, String param, String value) {
         try {
+            Span aciveSpan = tracer.activeSpan();
             HttpUrl url = new HttpUrl.Builder().scheme("http").host("localhost").port(port).addPathSegment(path)
                     .addQueryParameter(param, value).build();
             Request.Builder requestBuilder = new Request.Builder().url(url);
 
-            Tags.SPAN_KIND.set(tracer.activeSpan(), Tags.SPAN_KIND_CLIENT);
-            Tags.HTTP_METHOD.set(tracer.activeSpan(), "GET");
-            Tags.HTTP_URL.set(tracer.activeSpan(), url.toString());
-            tracer.inject(tracer.activeSpan().context(), Format.Builtin.HTTP_HEADERS, new RequestBuilderCarrier(requestBuilder));
+            Tags.SPAN_KIND.set(aciveSpan, Tags.SPAN_KIND_CLIENT);
+            Tags.HTTP_METHOD.set(aciveSpan, "GET");
+            Tags.HTTP_URL.set(aciveSpan, url.toString());
+            tracer.inject(aciveSpan.context(), Format.Builtin.HTTP_HEADERS, new RequestBuilderCarrier(requestBuilder));
 
             Request request = requestBuilder.build();
             Response response = client.newCall(request).execute();
             if (response.code() != 200) {
+                Tags.ERROR.set(aciveSpan, true);
+                Tags.HTTP_STATUS.set(aciveSpan, response.code());
+                aciveSpan.log(ImmutableMap.of("event", "error", "value", "Response code not 200"));
```

Comment:
* status code tag should be set regardless of whether it is 200 or not
* since this already throws exception, I just handle it in the catch, in which you already mark the span as erroneous

Reply:
* Ok, moving http_status code tag outside if block.
* The catch block handles one specific error type; Are you suggesting to make it more generic?

Reply:
basically this
```
            Tags.HTTP_STATUS.set(activeSpan, response.code());
            if (response.code() != 200) {
                throw new RuntimeException("Bad HTTP result: " + response);
            }
            return response.body().string();
        } catch (Exception e) {
            Tags.ERROR.set(activeSpan, true);
            activeSpan.log(ImmutableMap.of("event", "error", "error.object", e));
            throw new RuntimeException(e);
        }
```

Reply:
Ok. Will make exception handling generic.